penty/penty.py

```diff
@@ -840,10 +840,11 @@ def _bounded_attr(self, self_set, self_ty, attr):
 
         # propagate constexpr-ness
         if issubclass(func, CFT) and issubclass(self_ty, Cst):
-            return CFT[lambda *p: func(self_ty, *p),
-                       lambda *p: func.__args__[1](self_ty.__args__[0], *p)]
+            return CFT[lambda *p, **kwargs: func(self_ty, *p, **kwargs),
+                       lambda *p, **kwargs: func.__args__[1](
+                           self_ty.__args__[0], *p, **kwargs)]
         else:
-            return FT[lambda *p: func(self_ty, *p)]
+            return FT[lambda *p, **kwargs: func(self_ty, *p, **kwargs)]
```

Comment:
should this be a `MethodType` ?

Reply:
Atually no. A `MethodType` is an unbound method that will check that its first argument (`self`) has the appropriate  type. Here when we create this wrapper, the first argument is bound. When it will be called, the first argument will not be `self` but the actual first argument of the method call.